relay-general/src/store/geo.rs

```diff
@@ -54,6 +54,11 @@ impl GeoIpLookup {
                     .as_ref()
                     .and_then(|city| Some(city.names.as_ref()?.get("en")?.to_string())),
             ),
+            subdivisions: Annotated::from(
+                city.subdivisions
+                    .as_ref()
+                    .and_then(|subdivisions| Some(subdivisions.names.as_ref()?.get("en")?.to_string())),
+            ),
```

Comment:
this is failing because `subdivisions` is a list like this one:
```
"subdivisions": [
    {
      "geoname_id": 5332921,
      "iso_code": "CA",
      "names": {
        "de": "Kalifornien",
        "en": "California",
        "es": "California",
        "fr": "Californie",
        "ja": "カリフォルニア",
        "ru": "Калифорния",
        "zh-CN": "加州"
      }
    }
  ]
```
but I'm not too sure how to grab first element in Rust

Reply:
something like 

```
city.subdivisions.as_ref().and_then(|subdivisions| {
            subdivisions.get(0).and_then(|subdivision| {
                subdivision.names.as_ref().and_then(|subdivision_names| {
                    subdivision_names
                        .get("en")
                        .cloned()
                        .map(|subdivision_name| subdivision_name.to_string())
                })
            })
        });
```
i think?
i do wonder how often the multi subdivision case comes up and if its worth storing an array of them instead

Reply:
it's probably fine just to take the first (top level as i understand it) subdivision.

Reply:
based on this [example](https://dev.maxmind.com/geoip/docs/web-services/responses) it looks like it's always a list

Reply:
I was gonna grab the first element but rust wasn't happy about how I did it. I'll try your suggestion

Reply:
also changed it `subdivisions` -> `subdivision`